### DISCUSSION TITLE
How to get `PhpParser\Node\FunctionLike` from `PhpParser\Node\Expr\FuncCall`?

### DISCUSSION THREAD

**Discussion by Wirone** [rectorphp/rector#7037] [Q&A] [answered]
I'm struggling with inferring return type from `FuncCall` (function invoked as method's argument). My scenario:

- rector bound to `MethodCall` type
- if argument passed to method is a function call (argument is a `CallLike` instance) I want to determine possible return type
- if return type is `float` I want to refactor code

Right now I get `TypeError: Argument 2 passed to Rector\TypeDeclaration\TypeInferer\ReturnTypeInferer::isAutoImportWithFullyQualifiedReturn() must implement interface PhpParser\Node\FunctionLike, instance of PhpParser\Node\Expr\FuncCall given` when I do `$this->returnTypeInferer->inferFunctionLike($node);` (`$node` is `CallLike`).

I've tried to look at existing rectors but can't find any tip.. Maybe anybody can point me to solution 🙂

**Comment by TomasVotruba** [comment, marked as answer]
The `AstResolver` might help :+1:

**Reply by Wirone** [reply]
That's it! Thank you very much @TomasVotruba 🍻
